### ISSUE TITLE
[BUG] Failed to get backup-name backup status on unknown replica

### ISSUE THREAD

**Issue by AdamJSoftware** [longhorn/longhorn#4969]
## To Reproduce

Simple recurring backup

## Expected behavior
Backup properly completes

## Log or Support bundle
Following error message: Failed backup backup-3e15a3876d804aab has been deleted: proxyServer=10.42.0.222:8501 destination=10.42.0.222:10003: failed to get backup-3e15a3876d804aab backup status: rpc error: code = Unknown desc = failed to get backup-3e15a3876d804aab backup status on unknown replica tcp://10.42.0.221:10210

## Environment

 - Longhorn version: 1.3.2

## Additional context
it's a single node cluster, maybe that's why?

**Comment by derekbit** [comment]
Can you check the instance-manager log?

**Comment by tommy04062019** [comment]
> Can you check the instance-manager log?

@derekbit see u again:)), I also get this error when doing backup. Pls help to debug, log is attached below
[engine-manager.log](https://github.com/longhorn/longhorn/files/10297982/engine-manager.log)

**Comment by derekbit** [comment]
@tommy04062019 
Can you provide the log of the volume's instance-manager-r,  volume name and timing as well? 
I will check it. Thank you.

**Comment by tommy04062019** [comment]
> @tommy04062019 Can you provide the log of the volume's instance-manager-r, volume name and timing as well? I will check it. Thank you.

can u sent me ur email, my bundle is overlimit to upload :(

**Comment by derekbit** [comment]
@tommy04062019 
You can send to longhorn-support-bundle@suse.com. Thank you.

**Comment by tommy04062019** [comment]
> @tommy04062019 You can send to [longhorn-support-bundle@suse.com](mailto:longhorn-support-bundle@suse.com). Thank you.

check mail pls, I sent u, Thank u

**Comment by derekbit** [comment]
@tommy04062019 Can you provide the volume name?

**Comment by derekbit** [comment]
@tommy04062019 
I assume the failed backup happened on the volume `pvc-635abf58-50ff-4b39-a19a-6428d42c6ffb`.
```
creationtimestamp: "2022-12-24T04:13:30Z"

...

  message: 'Snapshot fcce053e-8d68-4f97-85af-e2523ff43e73 backup backup-64205d7c00dc4103
    label map[KubernetesStatus:{"pvName":"pvc-635abf58-50ff-4b39-a19a-6428d42c6ffb","pvStatus":"Bound","namespace":"minhbt-test","pvcName":"rwx-test","lastPVCRefAt":"","workloadsStatus":[{"podName":"rwx-test-5d98bd9fff-5kxsc","podStatus":"Running","workloadName":"rwx-test-5d98bd9fff","workloadType":"ReplicaSet"},{"podName":"rwx-test-5d98bd9fff-g8kvf","podStatus":"Running","workloadName":"rwx-test-5d98bd9fff","workloadType":"ReplicaSet"},{"podName":"rwx-test-5d98bd9fff-gwtbb","podStatus":"Running","workloadName":"rwx-test-5d98bd9fff","workloadType":"ReplicaSet"}],"lastPodRefAt":""}
    app:mysql longhorn.io/volume-access-mode:rwx]: proxyServer=10.42.96.233:8501 destination=10.42.50.194:10000:
    failed to get backup-64205d7c00dc4103 backup status: rpc error: code = Unknown
    desc = failed to get backup-64205d7c00dc4103 backup status on unknown replica
```

`SIGSEGV` happened when executing the backup job. This is related to https://github.com/longhorn/longhorn/issues/1768. It is fied in the upcoming v1.4.0 ([comment](https://github.com/longhorn/longhorn/issues/1768#issuecomment-1358751822)).

```
2022-12-24T04:13:29.166860937Z [pvc-635abf58-50ff-4b39-a19a-6428d42c6ffb-r-da5d9357] [signal SIGSEGV: segmentation violation code=0x1 addr=0x47 pc=0x7fb15428a341]
2022-12-24T04:13:29.166955026Z 
2022-12-24T04:13:29.166969607Z runtime stack:
2022-12-24T04:13:29.166978972Z runtime.throw(0xef4927, 0x2a)
2022-12-24T04:13:29.167067926Z 	/usr/local/go/src/runtime/panic.go:1116 +0x72
2022-12-24T04:13:29.167104203Z runtime.sigpanic()
2022-12-24T04:13:29.167111824Z 	/usr/local/go/src/runtime/signal_unix.go:701 +0x46a
2022-12-24T04:13:29.167118713Z 
2022-12-24T04:13:29.167131112Z goroutine 169070 [syscall]:
2022-12-24T04:13:29.167138571Z runtime.cgocall(0xc09a70, 0xc0004925c0, 0xc000ab40b8)
2022-12-24T04:13:29.167146115Z 	/usr/local/go/src/runtime/cgocall.go:133 +0x5b fp=0xc000492590 sp=0xc000492558 pc=0x403f7b
2022-12-24T04:13:29.167173067Z net._C2func_getaddrinfo(0xc00061ab20, 0x0, 0xc00010eae0, 0xc000ab40b8, 0x0, 0x0, 0x0)
2022-12-24T04:13:29.167184018Z 	_cgo_gotypes.go:92 +0x55 fp=0xc0004925c0 sp=0xc000492590 pc=0x6a3e65
2022-12-24T04:13:29.167192882Z net.cgoLookupIPCNAME.func1(0xc00061ab20, 0x14, 0x14, 0xc00010eae0, 0xc000ab40b8, 0xc000492638, 0xc0004926a0, 0x6a7242)
2022-12-24T04:13:29.167200410Z 	/usr/local/go/src/net/cgo_unix.go:161 +0xd2 fp=0xc000492608 sp=0xc0004925c0 pc=0x6a9982
2022-12-24T04:13:29.167208080Z net.cgoLookupIPCNAME(0xecb9fb, 0x3, 0xc00061aae0, 0x13, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
2022-12-24T04:13:29.167216786Z 	/usr/local/go/src/net/cgo_unix.go:161 +0x183 fp=0xc000492718 sp=0xc000492608 pc=0x6a52f3
2022-12-24T04:13:29.167267032Z net.cgoIPLookup(0xc00068f980, 0xecb9fb, 0x3, 0xc00061aae0, 0x13)
2022-12-24T04:13:29.167276666Z 	/usr/local/go/src/net/cgo_unix.go:218 +0x67 fp=0xc0004927b8 sp=0xc000492718 pc=0x6a5a27
2022-12-24T04:13:29.167283464Z runtime.goexit()
2022-12-24T04:13:29.167291014Z 	/usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc0004927c0 sp=0xc0004927b8 pc=0x4653c1
2022-12-24T04:13:29.167298296Z created by net.cgoLookupIP
2022-12-24T04:13:29.167305216Z 	/usr/local/go/src/net/cgo_unix.go:228 +0xc7
2022-12-24T04:13:29.167311872Z 
2022-12-24T04:13:29.167352547Z goroutine 1 [IO wait]:
2022-12-24T04:13:29.167365696Z internal/poll.runtime_pollWait(0x7fb15c223088, 0x72, 0x0)
2022-12-24T04:13:29.167373144Z 	/usr/local/go/src/runtime/netpoll.go:203 +0x55
2022-12-24T04:13:29.167380077Z internal/poll.(*pollDesc).wait(0xc000032698, 0x72, 0x0, 0x0, 0xecea60)
2022-12-24T04:13:29.167387234Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
2022-12-24T04:13:29.167393856Z internal/poll.(*pollDesc).waitRead(...)
2022-12-24T04:13:29.167400873Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:92
2022-12-24T04:13:29.167408531Z internal/poll.(*FD).Accept(0xc000032680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
2022-12-24T04:13:29.167438346Z 	/usr/local/go/src/internal/poll/fd_unix.go:384 +0x1d4
2022-12-24T04:13:29.167446811Z net.(*netFD).accept(0xc000032680, 0x37e11d600, 0x0, 0x0)
2022-12-24T04:13:29.167452924Z 	/usr/local/go/src/net/fd_unix.go:238 +0x42
2022-12-24T04:13:29.167460064Z net.(*TCPListener).accept(0xc0003b62c0, 0xc000458090, 0xc000458098, 0x18)
2022-12-24T04:13:29.167470292Z 	/usr/local/go/src/net/tcpsock_posix.go:139 +0x32
2022-12-24T04:13:29.167477741Z net.(*TCPListener).Accept(0xc0003b62c0, 0xf111e0, 0xc000001c80, 0x1155c20, 0xc000a5c0a0)
2022-12-24T04:13:29.167485807Z 	/usr/local/go/src/net/tcpsock.go:261 +0x64
2022-12-24T04:13:29.167492775Z google.golang.org/grpc.(*Server).Serve(0xc000001c80, 0x1149560, 0xc0003b62c0, 0x0, 0x0)
2022-12-24T04:13:29.167523957Z 	/go/src/github.com/longhorn/longhorn-engine/vendor/google.golang.org/grpc/server.go:587 +0x210
2022-12-24T04:13:29.167533571Z github.com/longhorn/longhorn-engine/app/cmd.startSyncAgent(0xc0000ff080, 0xeccbc3, 0x0)
2022-12-24T04:13:29.167541123Z 	/go/src/github.com/longhorn/longhorn-engine/app/cmd/sync_agent.go:89 +0x503
2022-12-24T04:13:29.167548283Z github.com/longhorn/longhorn-engine/app/cmd.SyncAgentCmd.func1(0xc0000ff080)
2022-12-24T04:13:29.167555602Z 	/go/src/github.com/longhorn/longhorn-engine/app/cmd/sync_agent.go:40 +0x2f
2022-12-24T04:13:29.167562823Z github.com/urfave/cli.HandleAction(0xd65560, 0xf10b48, 0xc0000ff080, 0xc0000ff080, 0x0)
2022-12-24T04:13:29.167570234Z 	/go/src/github.com/longhorn/longhorn-engine/vendor/github.com/urfave/cli/app.go:525 +0x7c
2022-12-24T04:13:29.167578312Z github.com/urfave/cli.Command.Run(0xed1e84, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xee9b64, ...)
2022-12-24T04:13:29.167621728Z 	/go/src/github.com/longhorn/longhorn-engine/vendor/github.com/urfave/cli/command.go:174 +0x56e
2022-12-24T04:13:29.167631518Z github.com/urfave/cli.(*App).Run(0xc0001f61c0, 0xc000032080, 0x8, 0x8, 0x0, 0x0)
2022-12-24T04:13:29.167638544Z 	/go/src/github.com/longhorn/longhorn-engine/vendor/github.com/urfave/cli/app.go:276 +0x7ae
2022-12-24T04:13:29.167645845Z main.longhornCli()
2022-12-24T04:13:29.167653278Z 	/go/src/github.com/longhorn/longhorn-engine/main.go:126 +0xaa3
2022-12-24T04:13:29.167660582Z main.main()
2022-12-24T04:13:29.167693047Z 	/go/src/github.com/longhorn/longhorn-engine/main.go:33 +0x13b
2022-12-24T04:13:29.167702301Z 
```

**Comment by tommy04062019** [comment]
> comment

Great, I will check it again with v1.4.0. Thank u:)

**Comment by tommy04062019** [comment]
@derekbit The backup made successfully in Longhorn v1.4.0 now, waiting for release, thank

**Comment by derekbit** [comment]
@tommy04062019  Awesome! Thanks for the feedback.

**Comment by innobead** [comment]
Closing first because the related cause has been fixed in 1.4.0, but feel free to reopen if @tommy04062019 @AdamJSoftware still encounters this issue.
